mono/mini/exceptions-amd64.c

```diff
@@ -71,6 +72,125 @@ static LONG CALLBACK seh_unhandled_exception_filter(EXCEPTION_POINTERS* ep)
 }
 
 /*
+* mono_win32_get_handle_stackoverflow (void):
+*
+* Returns a pointer to a method which restores the current context stack
+* and calls handle_exceptions, when done restores the original stack.
+*/
+static gpointer
+mono_win32_get_handle_stackoverflow (void)
+{
+	static guint8 *start = NULL;
+	guint8 *code;
+
+	if (start)
+		return start;
+
+	/* restore_contect (void *sigctx) */
+	start = code = mono_global_codeman_reserve (128);
+
+	/* save context into r12 */
+	amd64_mov_reg_reg (code, AMD64_R12, AMD64_RCX, 8);
+
+	/* move current stack into rdi for later restore */
+	amd64_mov_reg_reg (code, AMD64_RDI, AMD64_RSP, 8);
+
+	/* use the new freed stack from sigcontext */
+	amd64_mov_reg_membase (code, AMD64_RSP, AMD64_RCX, (MONO_STRUCT_OFFSET (CONTEXT, Rsp)), 8);
+
+	/* reserve 32 bytes of stack space for Win64 calling convention */
+	amd64_alu_reg_imm (code, X86_SUB, AMD64_RSP, 32);
+
+	/* get the current domain */
+	amd64_mov_reg_imm (code, AMD64_R11, mono_domain_get);
+	amd64_call_reg (code, AMD64_R11);
+
+	/* pop 32 bytes of stack space for Win64 calling convention */
+	amd64_alu_reg_imm (code, X86_ADD, AMD64_RSP, 32);
+
+	/* get stack overflow exception from domain object */
+	amd64_mov_reg_membase (code, AMD64_RAX, AMD64_RAX, G_STRUCT_OFFSET (MonoDomain, stack_overflow_ex), 8);
+
+	/* reserve 32 bytes of stack space for Win64 calling convention */
+	amd64_alu_reg_imm (code, X86_SUB, AMD64_RSP, 32);
+
+	/* call mono_arch_handle_exception (sctx, stack_overflow_exception_obj, FALSE) */
+	amd64_mov_reg_imm (code, AMD64_R8, 0);
+	amd64_mov_reg_reg (code, AMD64_RDX, AMD64_RAX, 8);
+	amd64_mov_reg_reg (code, AMD64_RCX, AMD64_R12, 8);
+	amd64_mov_reg_imm (code, AMD64_R11, mono_arch_handle_exception);
+	amd64_call_reg (code, AMD64_R11);
+
+	/* pop 32 bytes of stack space for Win64 calling convention */
+	amd64_alu_reg_imm (code, X86_ADD, AMD64_RSP, 32);
+
+	/* restore the SEH handler stack */
+	amd64_mov_reg_reg (code, AMD64_RSP, AMD64_RDI, 8);
+
+	/* return */
+	amd64_ret (code);
+
+	g_assert ((code - start) < 128);
+
```

Comment:
x86 version of this function have two function calls in the end. Same functions are used in other codegen methods in this source file. Maybe they should be included here as well to keep consistency?